packages/eslint-plugin/docs/rules/promise-function-async.mdx

```diff
@@ -9,16 +9,9 @@ import TabItem from '@theme/TabItem';
 >
 > See **https://typescript-eslint.io/rules/promise-function-async** for documentation.
 
-Ensures that each function is only capable of:
+Ensures that functions returning promises can only return rejected promises when an error is encountered and cannot throw. This simplifies the API for callers by making it so that they only need to handle errors in a single manner.
```

Comment:
🤔 I don't think this is better at explaining the rule. At its core, `promise-function-async` enforces that _"any function or method that returns a Promise to be marked async"_. This explanation is a rewording of the technical backing but both are pretty in-the-weeds.

I think I'd rather take this on myself as part of #8088 actually. Unless @kirkwaiblinger do you have specific wording you wanted to work with?

Reply:
I agree that this is in need of more significant overhaul from a team member 👍